text/0000-support-single-implementation.md

```diff
@@ -0,0 +1,131 @@
+
+Particular exporter *implementations* may support multiple data types, but are not required to; multi-purpose exporters must still be registered with OpenTelemetry as though they were single-purpose.
+
+### Out of the box behaviour
+
+As mentioned above, OpenTelemetry should discard data for which the user has not explicitly provided an exporter.
+
+This may be implemented through some sort of default no-op exporter, null check, "optional" exporter that defaults to "none", or whatever else is idiomatic to the particular language.
+
+This enables the default behaviour to:
+
+* Not error due to the lack of a user-provided exporter
+* Be lightweight, i.e., have minimal memory or CPU overhead
+
+### Supporting third-party needs
+
+Most third-party requirements should be addressable through either exporters or by extending the core API, depending on the particular requirement.
+
+For example, adding an extra ID to every span could be done by the exporter, whereas allowing the user to add attributes to spans must be done through the API itself.
+
+Exporters could possibly be extended to some sort of callback for relevant events (particularly span creation).
+
+### Migrating from the current architecture
+
+To the extent that the current architecture has already been implemented, a migration path may look something like:
+
+1. Start with the "minimal implementation"
+1. Send data to the particular event's exporter
+1. Make the "SDK" conform to the new exporter implementation
+1. Break out "transformer" logic from the SDK
+
+## Trade-offs and mitigations
+
+### Providing enough flexibility for vendors
+
+The biggest risk with this RFC is probably the question of whether it supports the needed flexibility.
+
+For potential future third-party needs that cannot be addressed with an exporter (or even a hook-supplemented exporter), the solution is likely to **add the functionality into the API** - particularly where multiple parties have the same or a similar need.
+
+See, for example, the [request for "transaction" support in OpenCensus](https://github.com/census-instrumentation/opencensus-specs/issues/229). A feature like this would be a strong candidate for getting added directly into the OpenTelemetry core.
+
+### Migration costs
+
+Given that OpenTelemetry is still very young and not yet production-ready, migration costs should be relatively small for both end users and implementors - particularly to the extent that the ambiguity between the API and SDK has slowed down implementation.
+
+## Prior art and alternatives
+
+### Prior art
+
+The three main prior art examples are OpenTracing, OpenCensus, and the existing OpenTelemetry architecture.
+
+**OpenTracing** handles export functionality by having each vendor (or other party) reimplement the whole API. While this maximizes the flexibility for vendors, it also means that everyone must redo a lot of the same logic - including any logic that may be considered vendor-agnostic.
```

Comment:
This is not a *requirement* of OpenTracing – it's feasible to create a high-quality, general-purpose implementation that vendors may choose (or not-choose) to extend. We did not have enough time/staffing to do that in OpenTracing out of the gate, but it was (and remains) an option.